NEWS.md

```diff
@@ -905,6 +905,11 @@ Deprecated or removed
   * `map` on dictionaries previously operated on `key=>value` pairs. This behavior is deprecated,
     and in the future `map` will operate only on values ([#5794]).
 
+  * Previously, broadcast defaulted to treating its arguments as scalars if they were not
```

Comment:
It would be useful to add a sentence for package developers mentioning `broadcastable`, just like the deprecation does.